Library/Homebrew/cmd/update-bash.sh

```diff
@@ -21,12 +21,17 @@ which_git() {
 }
 
 git_init_if_necessary() {
+  set -e
+  trap '{ rm -rf .git; exit 1; }' EXIT
+
   if [[ ! -d ".git" ]]
   then
     git init -q
     git config --bool core.autocrlf false
     git config remote.origin.url https://github.com/Homebrew/homebrew.git
     git config remote.origin.fetch "+refs/heads/*:refs/remotes/origin/*"
+    git fetch origin
+    git reset --hard origin/master
```

Comment:
Sorry, I checked again and yeh: it's present there. I had a think and the reason this isn't needed is because we already do this `fetch` (https://github.com/xu-cheng/homebrew/blob/update-bash-git-init/Library/Homebrew/cmd/update-bash.sh#L309-L310) and `reset` (https://github.com/xu-cheng/homebrew/blob/update-bash-git-init/Library/Homebrew/cmd/update-bash.sh#L187) later so you're duplicating the call here.